client/app/src/js/app.js

```diff
@@ -78,7 +77,17 @@ function config($urlRouterProvider, $stateProvider, $locationProvider, $httpProv
       url: '/screenshot/:screenshotId',
       templateUrl: '/views/screenshot.html',
       controller: 'ScreenshotController',
-      controllerAs: 'screenshotCtrl'
+      controllerAs: 'screenshotCtrl',
+      resolve: {
+        screenshot: function($stateParams, $window, Screenshot) {
```

Comment:
This would be a decent place for some quick documentation. A one-liner like "get screenshot based on ID before loading view, inject into controller" or something like that. My $0.02